qa/tasks/cephfs/test_admin.py

```diff
@@ -324,6 +324,76 @@ def test_fs_new_with_specific_id_fails_already_in_use(self):
         else:
             self.fail("expected creating file system with ID already in use to fail")
 
+    def test_fs_new_with_already_in_use_metadata_and_data_pool(self):
+        """
+        That creating file system with metadata and data pool which is already in use.
+        """
+
+        # create first data pool, metadata pool and add with filesystem
+        first_fs = "first_fs"
+        first_metadata_pool = "first_metadata_pool"
+        first_data_pool = "first_data_pool"
+        self.fs.mon_manager.raw_cluster_cmd('osd', 'pool', 'create', first_metadata_pool)
+        self.fs.mon_manager.raw_cluster_cmd('osd', 'pool', 'create', first_data_pool)
+        self.fs.mon_manager.raw_cluster_cmd('fs', 'new', first_fs, first_metadata_pool, first_data_pool)
+
+        # create second data pool, metadata pool and add with filesystem
+        second_fs = "second_fs"
+        second_metadata_pool = "second_metadata_pool"
+        second_data_pool = "second_data_pool"
+        self.fs.mon_manager.raw_cluster_cmd('osd', 'pool', 'create', second_metadata_pool)
+        self.fs.mon_manager.raw_cluster_cmd('osd', 'pool', 'create', second_data_pool)
+        self.fs.mon_manager.raw_cluster_cmd('fs', 'new', second_fs, second_metadata_pool, second_data_pool)
+
+        third_fs = "third_fs"
+
+        # try to create new fs 'third_fs' with following configuration
+        # metadata pool -> 'second_metadata_pool'
+        # data pool -> 'second_data_pool'
+        # Expecting EINVAL exit status because 'second_metadata_pool' and 'second_data_pool'
+        # is already in use with 'second_fs'
+        try:
+            self.fs.mon_manager.raw_cluster_cmd('fs', 'new', third_fs, second_metadata_pool, second_data_pool)
```

Comment:
Yeah, this test case looks good.

You'd better to use only either old metadata pool or old data pool to test this and the following ones. Because we don't know whether both metadata and data pools' checks are working or only one of them could work.

Reply:
Done